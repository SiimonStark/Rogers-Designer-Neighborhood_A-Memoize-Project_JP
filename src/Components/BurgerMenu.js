import React, { Component } from 'react';

class BurgerMenu extends Component {
  constructor(props) {
    super(props)
    this.state = {
    }
  }
  reloadPage = () => {
    window.location.reload(true);
  }
  displayAlert = () => {
    alert('Feature Coming Soon');
  }

  render() {
    
    return (
      <aside className="BurgerMenu" onMouseLeave={this.props.mouseLeave}>
        <ul>
          <li onClick={this.reloadPage}>Reset Game</li>
          <li onClick={this.displayAlert}>Skip Prompt</li>
          <li onClick={this.displayAlert}>Save Game</li>
          <li>
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/grid"
              target="blank"
            >
              Docs: Grid
            </a>
          </li>
          <li>
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/flex"
              target="blank"
            >
              Docs: Flex
            </a>
          </li>
        </ul>
      </aside>
    );
  }
}

export default BurgerMenu;