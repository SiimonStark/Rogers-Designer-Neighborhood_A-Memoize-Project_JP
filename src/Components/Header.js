import React, { Component } from 'react';
import BurgerMenu from './BurgerMenu';

class Header extends Component {
  constructor() {
    super()
    this.state = {
      showBurgerMenu: false
    }
  }
  showBurgerMenu = () => {
    this.state.showBurgerMenu
      ? this.setState({showBurgerMenu: false})
      : this.setState({showBurgerMenu: true});
  }
  render() {
    let menu;
    this.state.showBurgerMenu ? menu = <BurgerMenu/> : menu = null;
    return (
      <header>
      <div className="sky"></div>
      <div className="logo">
      </div>
        <nav>
          <button onClick={this.showBurgerMenu}>MENU</button>
          {menu}
        </nav>
      </header>
    );
  }
}

export default Header;