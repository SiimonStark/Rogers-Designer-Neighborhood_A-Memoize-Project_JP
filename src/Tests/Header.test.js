import React from 'react';
import Header from '../Components/Header';
import BurgerMenu from '../Components/BurgerMenu';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header />)
  });

  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match conditional render Snapshot', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state('showBurgerMenu')).toEqual(false);
  });

  it('should toggle showBurgerMenu onClick &  showBurgerMenu() is invoked', () => {
    expect(wrapper.state("showBurgerMenu")).toEqual(false);
    wrapper.find('button').simulate('click');
    expect(wrapper.state("showBurgerMenu")).toEqual(true);
    wrapper.find("button").simulate("click");
    expect(wrapper.state("showBurgerMenu")).toEqual(false);
  });

  it('should toggle showBurgerMenu when mouseLeave() is invoked', () => {
    let burger = shallow(<BurgerMenu
      mouseLeave={wrapper.mouseLeave} />);
    
    expect(wrapper.state("showBurgerMenu")).toEqual(false);
    burger.find("aside").simulate("mouseLeave");
    wrapper.instance().showBurgerMenu();
    expect(wrapper.state("showBurgerMenu")).toEqual(true);
    burger.find("aside").simulate("mouseLeave");
    wrapper.instance().showBurgerMenu();
    expect(wrapper.state("showBurgerMenu")).toEqual(false);
  });
})