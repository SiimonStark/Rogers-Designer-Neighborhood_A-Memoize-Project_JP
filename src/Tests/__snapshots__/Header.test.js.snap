// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match Snapshot 1`] = `
<header>
  <div
    className="sky"
  />
  <div
    className="logo"
  />
  <nav>
    <button
      onClick={[Function]}
    >
      MENU
    </button>
  </nav>
</header>
`;

exports[`Header should match conditional render Snapshot 1`] = `
<header>
  <div
    className="sky"
  />
  <div
    className="logo"
  />
  <nav>
    <button
      onClick={[Function]}
    >
      MENU
    </button>
    <BurgerMenu
      mouseLeave={[Function]}
    />
  </nav>
</header>
`;
